app/services/session_encryptor.rb

```diff
@@ -8,15 +10,23 @@ def self.build_user_access_key
     build_user_access_key
   end
 
-  # rubocop:disable Security/MarshalLoad
   def self.load(value)
     decrypted = encryptor.decrypt(value, user_access_key)
-    Marshal.load(::Base64.decode64(decrypted))
+
+    if decrypted.start_with?(MARSHAL_SIGNATURE)
+      Rails.logger.info 'Marshalled session found'
+      # rubocop:disable Security/MarshalLoad
+      Marshal.load(::Base64.decode64(decrypted)).tap do |decoded_value|
```

Comment:
assuming we will not need this forever, perhaps put a comment in that links to an issue re: removing this? (look at me, advocating code comments all over the place!)